### ISSUE TITLE
(현화님) 개인별 모델 baseline 구축 및 실험 결과 공유

### ISSUE THREAD

**Issue by nimnusmik** [AIBootcamp14/upstageailab-ml-competition-ml-5#72]
<html>
<body>
<h1>🏢 아파트 실거래가 예측 모델링 분석 및 버전별 비교</h1>
<p>본 이슈는 사용된 3가지의 주요 노트북 파일(<code>price_prediction4.ipynb</code>, <code>price_prediction5.ipynb</code>, <code>price_prediction6.ipynb</code>)의 모델링 과정을 분석하고, 데이터 전처리와 피처 엔지니어링, 모델 학습 방식 및 하이퍼파라미터 최적화 결과를 자세히 기록합니다.</p>
<hr>
<h2>📌 데이터 사용 현황</h2>
<ul>
<li><strong><code>price_prediction4.ipynb</code></strong>, <strong><code>price_prediction5.ipynb</code></strong>
<ul>
<li>사용 데이터: <code>data/processed/cleaned_data/train_clean.csv</code>, <code>test_clean.csv</code> (좌표 데이터 미포함)</li>
</ul>
</li>
<li><strong><code>price_prediction6.ipynb</code></strong>
<ul>
<li>사용 데이터: 좌표X, 좌표Y 추가된 데이터 (<code>data/processed/clean_data/train.csv</code>, <code>test.csv</code>)</li>
</ul>
</li>
</ul>
<hr>
<h2>🚀 모델링 세부 분석</h2>
<h3>📝 공통 사항</h3>
<ul>
<li>Target 변수 로그 변환 적용</li>
<li>시간 관련 특성 및 면적당 가격 특성, 상호작용 특성 생성</li>
<li>범주형 변수 Label Encoding 적용</li>
<li>최종 특성 선택을 위한 사전 모델 학습 진행 후 특성 개수 결정</li>
</ul>
<h3>🧾 price_prediction4.ipynb</h3>
<h3>🔧 데이터 전처리 및 피처 엔지니어링</h3>
<ul>
<li>시간 관련(<code>계약월_sin</code>, <code>계약월_cos</code>, <code>아파트나이</code>), 면적당 가격(<code>동별/구별_평균면적당가격</code>, <code>동별/구별_면적당가격편차</code>), 상호작용 특성(<code>면적_x_나이</code>, <code>면적_x_층</code>, <code>강남_x_면적</code>) 추가</li>
<li>결측치 처리 완료</li>
<li>최종 특성 개수: 28개 (최초 특성 중 10개 제외)</li>
</ul>
<h3>🎯 하이퍼파라미터 최적화 (Optuna)</h3>
<ul>
<li>
<p>각 Trial마다 5-Fold 교차 검증 수행으로 시간이 매우 오래 걸림</p>
</li>
<li>
<p>최적 RMSE: <code>0.08895</code></p>
</li>
<li>
<p>최적 파라미터:</p>
<pre><code class="language-json">{
    &quot;learning_rate&quot;: 0.04992,
    &quot;num_leaves&quot;: 142,
    &quot;max_depth&quot;: 9,
    &quot;subsample&quot;: 0.8427,
    &quot;colsample_bytree&quot;: 0.9822,
    &quot;lambda_l1&quot;: 0.00017,
    &quot;lambda_l2&quot;: 0.000038
}
</code></pre>
</li>
</ul>
<hr>
<h3>🧾 price_prediction5.ipynb</h3>
<h3>🔧 데이터 전처리 및 피처 엔지니어링</h3>
<ul>
<li>데이터 및 피처 엔지니어링 방식은 v4와 동일</li>
<li>최종 특성 개수: 28개 (최초 특성 중 10개 제외)</li>
</ul>
<h3>🎯 하이퍼파라미터 최적화 (Optuna)</h3>
<ul>
<li>
<p>전체 훈련 데이터의 20%를 미리 검증 데이터로 분리하여 단일 세트로 Optuna 최적화 수행 (속도 개선)</p>
</li>
<li>
<p>최적 RMSE: <code>0.09830</code></p>
</li>
<li>
<p>최적 파라미터:</p>
<pre><code class="language-json">{
    &quot;learning_rate&quot;: 0.04977,
    &quot;num_leaves&quot;: 150,
    &quot;max_depth&quot;: 23,
    &quot;subsample&quot;: 0.9128,
    &quot;colsample_bytree&quot;: 0.9368,
    &quot;lambda_l1&quot;: 0.1908,
    &quot;lambda_l2&quot;: 0.0012,
    &quot;objective&quot;: &quot;regression_l1&quot;,
    &quot;metric&quot;: &quot;rmse&quot;,
    &quot;n_estimators&quot;: 2000,
    &quot;verbosity&quot;: -1,
    &quot;seed&quot;: 42,
    &quot;device&quot;: &quot;cuda&quot;
}

</code></pre>
</li>
</ul>
<h3>📊 교차 검증 (K-Fold)</h3>
<ul>
<li>Fold 1 Log-RMSE: <code>0.0875</code></li>
<li>Fold 2 Log-RMSE: <code>0.0874</code></li>
<li>Fold 3Log-RMSE: <code>0.0891</code></li>
<li>Fold 4 Log-RMSE: <code>0.0879</code></li>
<li>Fold 5 Log-RMSE: <code>0.0886</code></li>
</ul>

<img width="1178" height="841" alt="Image" src="https://github.com/user-attachments/assets/07a8ec7a-bd9d-4503-9f12-649fa9f5325b" />

<img width="829" height="862" alt="Image" src="https://github.com/user-attachments/assets/35c75208-f96a-465d-b038-fdcc3d99ba83" />

<img width="880" height="554" alt="Image" src="https://github.com/user-attachments/assets/cbd94ef5-065f-4fc8-a0ca-1289aadad41e" />

<img width="861" height="554" alt="Image" src="https://github.com/user-attachments/assets/e4b82124-47e1-46cd-a14d-929d4f446ae6" />

<hr>
<h3>🧾 price_prediction6.ipynb</h3>
<h3>🔧 데이터 전처리 및 피처 엔지니어링</h3>
<ul>
<li>좌표X, 좌표Y 데이터 추가</li>
<li>K-평균 군집화(5개 특성 사용, 10개 군집 생성)로 <code>아파트군집</code> 생성</li>
<li>교통 관련 가중합 특성(<code>가중지하철</code>, <code>가중버스</code>) 추가</li>
<li>시간 관련(sin/cos), 면적당 가격, 상호작용 특성 추가</li>
<li>최종 특성 개수: 40개 (최초 특성 중 4개 제외)</li>
</ul>
<h3>🎯 하이퍼파라미터 최적화 (Optuna)</h3>
<ul>
<li>빠른 최적화 위해 데이터 시간순으로 마지막 20% 검증 데이터 분리하여 단일 검증 세트로 최적화</li>
<li>교차 검증 방식: TimeSeriesSplit로 변경</li>
<li>특성 선택 및 데이터 분리 완료</li>
<li>최적 파라미터: 아직 도출되지 않음 (진행 중)</li>
</ul>

<img width="1184" height="984" alt="Image" src="https://github.com/user-attachments/assets/cc3f72cc-c067-4f9b-b70d-b3565ebcc332" />

<img width="984" height="984" alt="Image" src="https://github.com/user-attachments/assets/ea3dd172-b52e-4e52-998b-9faf4c0a8483" />

<img width="984" height="584" alt="Image" src="https://github.com/user-attachments/assets/84e36955-dc99-4715-b5e3-98bb232686dd" />

<img width="982" height="584" alt="Image" src="https://github.com/user-attachments/assets/fb05f617-43cd-48fa-b17a-47bb24c55709" />

<img width="984" height="584" alt="Image" src="https://github.com/user-attachments/assets/2258ea8d-9d94-4bdd-9fa9-7765ae36a76e" />

<img width="851" height="529" alt="Image" src="https://github.com/user-attachments/assets/7ca4ee65-5bb4-434d-9a4e-95ec87986699" />

<hr>
<h2>🔄 버전별 차이점 요약</h2>

항목 | v4 | v5 | v6
-- | -- | -- | --
데이터 | 좌표 미포함 | 동일 | 좌표 데이터 추가
피처 공학 | 기본 통계/상호작용 | 동일 | K-Means 군집화, 교통 가중합 추가
검증 방법 | Stratified K-Fold | 동일 | TimeSeriesSplit
최적화 속도 | 느림 (5-Fold CV 수행) | 빠름 (단일 검증 세트) | 빠름 (단일 시계열 검증 세트)
최종 피처 수 | 28개 | 28개 | 40개

<hr>

</body>
</html>
